### ISSUE TITLE
Steps for renaming a snapshot

### ISSUE THREAD

**Issue by vb0** [gilbertchen/duplicacy#224]
What are the steps for renaming a snapshot? Will renaming the folder in whatever_storage\snapshots and in .duplicacy/preferences (id:) do ar is there something more involved?

**Comment by gilbertchen** [comment]
Yes, those two changes should do it.

**Comment by jt70471** [comment]
And I think .duplicacy/cache/whatever_storage/snapshots as well.

**Comment by vb0** [comment]
I did all changes and then list worked fine, did one backup worked fine too (well actually just to advance the revision). However next "list" would break with:

Failed to decrypt the file snapshots/new_name/1: cipher: message authentication failed

It seems that the snapshot names are also inside those 1,2,3... files.

If I remove 1 and 2 (I'm at rev 3 done with the new name) everything works (of course I don't have revs 1&2 available anymore and I presume there would be some orphan files left).

**Comment by vb0** [comment]
All right, I put 1 and 2 back to snapshots/old_name and as I was suspecting (I don't know precisely how it works but I've got the idea over a long drive...) and now I have:

Snapshot old_name revision 1
Snapshot old_name revision 2
Snapshot new_name revision 3
Snapshot new_name revision 4
....

I assume everything would be somehow consistent going forward and if I really want I can clean the old_name and revisions.

**Comment by gilbertchen** [comment]
Sorry I forgot that snapshots on an encrypted storage can't be easily renamed, because the snapshot files are encrypted with keys derived from their paths, so once they are renamed they can't be decrypted.

**Comment by schmitmd** [comment]
Is there going to be code implementation at some point that allows a rename (change of "id:" value) for an encrypted destination?  Is it relevant that some of us store all repositories in the same root folder (gcd://duplicacy_backups , for instance)?

**Comment by TheBestPessimist** [comment]
id is the same both for encrypted and without. just rename the 2 places as per OP.

**Comment by schmitmd** [comment]
Got confused by gilbertchen's last comment.  My bad!  Thx for clearing me up.

**Comment by gilbertchen** [comment]
No, renaming the snapshot directories only works for unencrypted storages.  For an encrypted storage, since the snapshot file is encrypted with a key derived from the path of the snapshot file, after renaming you won't be able to decrypt the snapshot file.

 This is more of a design flaw.  I plan to fix this during the encryption upgrade to support per-user encryption.

**Comment by TheBestPessimist** [comment]
Ah, my knowledge is bad :(
